experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/ApplicationMaster.scala

```diff
@@ -0,0 +1,193 @@
+import org.apache.hadoop.yarn.client.api.async.NMClientAsync
+import org.apache.hadoop.yarn.api.records.ContainerId
+import java.nio.ByteBuffer
+import org.apache.hadoop.yarn.api.records.ContainerStatus
+import java.util.concurrent.atomic.AtomicInteger
+import org.apache.hadoop.yarn.api.records.Container
+import scala.collection.JavaConverters._
+import org.apache.hadoop.yarn.api.records.NodeReport
+import org.apache.hadoop.yarn.api.records.ContainerExitStatus
+import org.apache.hadoop.yarn.api.records.ContainerLaunchContext
+import org.apache.hadoop.yarn.api.records.LocalResource
+import org.apache.hadoop.yarn.api.records.FinalApplicationStatus
+
+/**
+ * Yarn ApplicationMaster. 
+ */
+class ApplicationMaster(amArgs: ApplicationMasterArguments, yarnConf : YarnConfiguration) {
+
+import org.apache.gearpump.experiments.yarn.ApplicationMaster._
+
+	private val LOG: Logger = LogUtil.getLogger(getClass)  
+  private val ONE_SEC_INTERVAL = 1000
+  private var amRMClient: AMRMClientAsync[ContainerRequest] = _
+  private var nmCallbackHandler: NMCallbackHandler = _
+  private var nmClientAsync: NMClientAsync = _ 
+  private var rmCallbackHandler: RMCallbackHandler = _
+  
+  def getYarnConf = yarnConf
+  def getamArgs = amArgs
+  
+
+private def getAMRMClient(rmCallbackHandler: RMCallbackHandler): AMRMClientAsync[ContainerRequest] = {
+	val amrmClient:AMRMClientAsync[ContainerRequest] = AMRMClientAsync.createAMRMClientAsync(ONE_SEC_INTERVAL, rmCallbackHandler)
+	amrmClient.init(getYarnConf)
+	amrmClient.start
+	amrmClient
+}
+
+private def getNMClient(containerListener:NMCallbackHandler) = {
+	val nmClient = new NMClientAsyncImpl(containerListener)
+	nmClient.init(getYarnConf);
+	nmClient.start();
+	nmClient    
+}
+
+private def registerAm:RegisterApplicationMasterResponse = {
+	val response = amRMClient.registerApplicationMaster("", 0, "")
+
+			LOG.info(s"Max mem and vcores capabilities : " + 
+					response.getMaximumResourceCapability.getMemory + " and " + 
+					response.getMaximumResourceCapability.getVirtualCores)
+  response
+}
+
+private def requestContainers(registrationResponse: RegisterApplicationMasterResponse) {
+	val previousContainersCount = registrationResponse.getContainersFromPreviousAttempts().size()
+			LOG.info(s"Previous container count : $previousContainersCount")
+			val containersToRequestCount = amArgs.containers - previousContainersCount;
+  
+	for(x <- 1 to containersToRequestCount) {
+		    val containerRequest = createContainerRequest(amArgs.containerMemory, amArgs.containerVCores)
+				LOG.info(s"Requested container : $containerRequest")
+			  amRMClient.addContainerRequest(containerRequest)
+	}
+}
+
+
+def run:Boolean = {
+    LOG.info("Initializing application master")
+    initialize    
+    
+    LOG.info("Requesting containers")
+    requestContainers(registerAm)
+    
+    LOG.info("Stopping application master")
+    waitForContainers
+    waitForLauncherThreads
+    cleanUp 
+  }
+
+  private def initialize {
```

Comment:
All function should define a return value.

If there is no return, we should define it explicitly as

def initilaize: Unit =